client/ws-client/src/tests.rs

```diff
@@ -68,8 +68,12 @@ async fn method_call_with_wrong_id_kind() {
 	let client =
 		WsClientBuilder::default().id_format(IdKind::String).build(&uri).with_default_timeout().await.unwrap().unwrap();
 
-	let err: Result<String, Error> = client.request("o", rpc_params![]).with_default_timeout().await.unwrap();
-	assert!(matches!(err, Err(Error::RestartNeeded(e)) if e == "request ID=0 is not a pending call"));
+	match client.request::<String, _>("o", rpc_params![]).with_default_timeout().await.unwrap() {
```

Comment:
nit: Maybe `assert_matches!` cleans this up a bit, but since its part of testing I wouldn't worry too much about it

Reply:
I wanted to show that working with Error::RestartNeeded(Arc<Error>) can be awkward :P